src/runner/IntegrationsWithOtherPackages.jl

```diff
@@ -0,0 +1,143 @@
+module IntegrationsWithOtherPackages
+
+import ..workspace_info
+include("./Requires.jl/src/Requires.jl")
+
+"Attempts to find the MIME pair corresponding to the extension of a filename. Defaults to `text/plain`."
+function mime_fromfilename(filename)
+    # This bad boy is from: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+    mimepairs = Dict(".aac" => "audio/aac", ".bin" => "application/octet-stream", ".bmp" => "image/bmp", ".css" => "text/css", ".csv" => "text/csv", ".eot" => "application/vnd.ms-fontobject", ".gz" => "application/gzip", ".gif" => "image/gif", ".htm" => "text/html", ".html" => "text/html", ".ico" => "image/vnd.microsoft.icon", ".jpeg" => "image/jpeg", ".jpg" => "image/jpeg", ".js" => "text/javascript", ".json" => "application/json", ".jsonld" => "application/ld+json", ".mjs" => "text/javascript", ".mp3" => "audio/mpeg", ".mp4" => "video/mp4", ".mpeg" => "video/mpeg", ".oga" => "audio/ogg", ".ogv" => "video/ogg", ".ogx" => "application/ogg", ".opus" => "audio/opus", ".otf" => "font/otf", ".png" => "image/png", ".pdf" => "application/pdf", ".rtf" => "application/rtf", ".sh" => "application/x-sh", ".svg" => "image/svg+xml", ".tar" => "application/x-tar", ".tif" => "image/tiff", ".tiff" => "image/tiff", ".ttf" => "font/ttf", ".txt" => "text/plain", ".wav" => "audio/wav", ".weba" => "audio/webm", ".webm" => "video/webm", ".webp" => "image/webp", ".woff" => "font/woff", ".woff2" => "font/woff2", ".xhtml" => "application/xhtml+xml", ".xml" => "application/xml", ".xul" => "application/vnd.mozilla.xul+xml", ".zip" => "application/zip")
+    file_extension = getkey(mimepairs, '.' * split(filename, '.')[end], ".txt")
+    MIME(mimepairs[file_extension])
+end
+
+export handle_websocket_message, message_channel
+"Called directly (through Distributed) from the main Pluto process"
+function handle_websocket_message(message)
+    try
+        result = on_websocket_message(Val(Symbol(message[:module_name])), message[:body])
+        if result !== nothing
+            @warn "Integrations `on_websocket_message($(Symbol(message[:module_name])))` returned a value, but is expected to return `nothing`"
+        end
+    catch ex
+        bt = stacktrace(catch_backtrace())
+        @error "Dispatching integrations websocket message failed:" message=message exception=(ex, bt)
+    end
+    nothing
+end
+const message_channel = Channel{Any}(10)
+
+"""
+Do not call yourself! Or do, but don't say I didn't warn you.
+Integrations should implement this to capture incoming websocket messages.
+I force returning nothing, because returning might give you the idea that
```

Comment:
I think that the `::Nothing` on this method does not apply to the methods added by integrations, so we should typeassert somewhere else

Reply:
Sorry you already do that!